### ISSUE TITLE
sbt-updates fails with implementation is missing

### ISSUE THREAD

**Issue by JonathanKnight** [frugalmechanic/fm-sbt-s3-resolver#17]
scala.NotImplementedError: an implementation is missing
    at scala.Predef$.$qmark$qmark$qmark(Predef.scala:252)
    at fm.sbt.s3.Handler.openConnection(Handler.scala:24)

**Comment by tpunder** [comment]
Interesting... Is there any additional stack trace?  Can you share any "s3://" URL references you have in your build files?

I had to implement a URLStream handler in order to allow s3:// URLs to be created but there should be nothing calling openConnection on an s3:// URL.

**Comment by JonathanKnight** [comment]
Here is the stack trace
    scala.NotImplementedError: an implementation is missing
	at scala.Predef$.$qmark$qmark$qmark(Predef.scala:252)
	at fm.sbt.s3.Handler.openConnection(Handler.scala:24)
	at java.net.URL.openConnection(URL.java:972)
	at java.net.URL.openStream(URL.java:1038)
	at scala.xml.factory.XMLLoader$class.load(XMLLoader.scala:56)
	at scala.xml.XML$.load(XML.scala:57)
	at com.timushev.sbt.updates.MetadataLoaderFactory$$anonfun$download$1$$anonfun$apply$1.apply(MetadataLoader.scala:23)
	at com.timushev.sbt.updates.MetadataLoaderFactory$$anonfun$download$1$$anonfun$apply$1.apply(MetadataLoader.scala:21)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at scala.concurrent.impl.ExecutionContextImpl$$anon$3.exec(ExecutionContextImpl.scala:107)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)


The sbt-updates library is looking for data on any newer version of the library.

**Comment by tpunder** [comment]
Thanks.  I'll have to play around with the sbt-updates library and then figure out what it will take to fully implement `URLStreamHandler` and `URLConnection` for s3:// URLs.

**Comment by cosmincatalin** [comment]
:+1: This would be a greatly appreciated feature.

**Comment by rtfpessoa** [comment]
Hi @tpunder 

I am having the same issue here: https://github.com/alexarchambault/coursier/issues/164#issuecomment-221694155

Can you check if you know what might be the problem?

**Comment by newhoggy** [comment]
I'm also getting this issue:

```
java.lang.Exception: Exception during resolutionbor/sbt-avrora/scala_2.10/sbt_0.13/0.0.1/ivys/ivy.xml
        at coursier.Tasks$$anonfun$resolutionTask$1$$anonfun$28.apply(Tasks.scala:462)
s3://proat coursier.Tasks$$anonfun$resolutionTask$1$$anonfun$28.apply(Tasks.scala:462).1/ivys/ivy.xml.sha1
        at scalaz.$bslash$div.leftMap(Either.scala:97)
	at coursier.Tasks$$anonfun$resolutionTask$1.coursier$Tasks$$anonfun$$resolution$1(Tasks.scala:462)
	at coursier.Tasks$$anonfun$resolutionTask$1$$anonfun$apply$35.apply(Tasks.scala:508)
	at coursier.Tasks$$anonfun$resolutionTask$1$$anonfun$apply$35.apply(Tasks.scala:508)
	at scala.collection.mutable.MapLike$class.getOrElseUpdate(MapLike.scala:189)
	at scala.collection.mutable.AbstractMap.getOrElseUpdate(Map.scala:91)
	at coursier.Tasks$$anonfun$resolutionTask$1.apply(Tasks.scala:501)
	at coursier.Tasks$$anonfun$resolutionTask$1.apply(Tasks.scala:227)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: scala.NotImplementedError: an implementation is missing
	at scala.Predef$.$qmark$qmark$qmark(Predef.scala:252)
	at fm.sbt.s3.Handler.openConnection(Handler.scala:24)
	at java.net.URL.openConnection(URL.java:972)
	at coursier.Cache$.coursier$Cache$$urlConn$1(Cache.scala:317)
	at coursier.Cache$$anonfun$remote$1$1$$anonfun$apply$14$$anonfun$apply$15.apply(Cache.scala:451)
	at coursier.Cache$$anonfun$remote$1$1$$anonfun$apply$14$$anonfun$apply$15.apply(Cache.scala:446)
	at coursier.Cache$.coursier$Cache$$downloading(Cache.scala:195)
	at coursier.Cache$$anonfun$remote$1$1$$anonfun$apply$14.apply(Cache.scala:446)
	at coursier.Cache$$anonfun$remote$1$1$$anonfun$apply$14.apply(Cache.scala:446)
	at coursier.Cache$.coursier$Cache$$withLockFor(Cache.scala:164)
	at coursier.Cache$$anonfun$remote$1$1.apply(Cache.scala:445)
	at coursier.Cache$$anonfun$remote$1$1.apply(Cache.scala:445)
	at scalaz.concurrent.Task$.Try(Task.scala:386)
	at scalaz.concurrent.Task$$anonfun$apply$10.apply(Task.scala:286)
	at scalaz.concurrent.Task$$anonfun$apply$10.apply(Task.scala:286)
	at scalaz.concurrent.Future$$anonfun$apply$15$$anon$4.call(Future.scala:380)
	at scalaz.concurrent.Future$$anonfun$apply$15$$anon$4.call(Future.scala:380)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[error] (*:coursierResolution) java.lang.Exception: Exception during resolution
```

**Comment by mscharley** [comment]
Also getting this too. Is there any update on this?

```
scala.NotImplementedError: an implementation is missing
        at scala.Predef$.$qmark$qmark$qmark(Predef.scala:252)
        at fm.sbt.s3.Handler.openConnection(Handler.scala:24)
        at java.net.URL.openConnection(URL.java:979)
        at java.net.URL.openStream(URL.java:1045)
        at scala.xml.factory.XMLLoader$class.load(XMLLoader.scala:56)
        at scala.xml.XML$.load(XML.scala:57)
        at com.timushev.sbt.updates.MetadataLoaderFactory$$anonfun$download$1$$anonfun$apply$1.apply(MetadataLoader.scala:34)
        at com.timushev.sbt.updates.MetadataLoaderFactory$$anonfun$download$1$$anonfun$apply$1.apply(MetadataLoader.scala:25)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
        at scala.concurrent.impl.ExecutionContextImpl$$anon$3.exec(ExecutionContextImpl.scala:107)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

**Comment by tpunder** [comment]
I've just published version 0.10.0 which has initial support for Coursier via commit https://github.com/frugalmechanic/fm-sbt-s3-resolver/commit/1aa10983e3ba70eefef014fbdd57fb6a66c0a47f
